sdl_android/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java

```diff
@@ -47,7 +49,28 @@ public void parseRAIResponse(RegisterAppInterfaceResponse response){
 	 * @param capability
 	 */
 	public void setCapability(SystemCapabilityType systemCapabilityType, Object capability){
-		cachedSystemCapabilities.put(systemCapabilityType,capability);
+		cachedSystemCapabilities.put(systemCapabilityType, capability);
+		notifyListeners(systemCapabilityType, capability);
+	}
+
+	/**
+	 * Notify listners in the list about the new retrieved capability
+	 * @param systemCapabilityType
+	 * @param capability
+	 */
+	private void notifyListeners(SystemCapabilityType systemCapabilityType, Object capability) {
+		synchronized(LISTENER_LOCK){
+			CopyOnWriteArrayList<OnSystemCapabilityListener> listeners = onSystemCapabilityListeners.get(systemCapabilityType);
+			if(listeners != null && listeners.size() > 0) {
+				for (OnSystemCapabilityListener listener : listeners) {
+					if (capability != null) {
+						listener.onCapabilityRetrieved(capability);
+					} else {
+						listener.onError(String.format("capability: %s is null", systemCapabilityType.toString()));
```

Comment:
I'm not sure if this is really an error case. Some systems might not support all capabilities and some are not mandatory parameters so they might be null.

Reply:
Right, that makes sense. I will fix that.